### PR TITLE
Presence state fix lk

### PR DESCRIPTION
**Pull request by kleewho**
Refactor proposal

### CODE REVIEW

File: core/pubnub_coreapi.c [hunk 1]
```diff
@@ -60,29 +60,26 @@ static void check_if_default_channel_and_groups(pubnub_t* p,
...
 #endif /* PUBNUB_USE_AUTO_HEARTBEAT */
 
 
 enum pubnub_res pubnub_leave(pubnub_t* p, const char* channel, const char* channel_group)
 {
     enum pubnub_res rslt;
-    char const* prep_channel;
-    char const* prep_channel_group;
+    char const*     prep_channel;
+    char const*     prep_channel_group;
```
Comment (kleewho) [inline core/pubnub_coreapi.c:82]: I guess formatting whole file was a mistake :/

**Comment by budgetpreneur** [comment]
with the current version of code, I am getting sanitizer issues on Windows
Setting state...
pbcc_set_state_prep. REQUEST =/v2/presence/sub-key/demo/channel/hello_world,test_world/uuid/4bb201b4-d40b-4a24-8191-08c454ea21b7/data?pnsdk=Windows-PubNub-C-core%2F3.4.1&channel-group=cg1,cg2&auth=danaske&state=%7B%22x%22%3A5%7D
=================================================================
==8088==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x0350308d at pc 0x00508269 bp 0x003cfb24 sp 0x003cfb18
READ of size 78 at 0x0350308d thread T0
    #0 0x508281 in __asan_wrap_strlen D:\a01\_work\10\s\src\vctools\crt\asan\llvm\compiler-rt\lib\sanitizer_common\sanitizer_common_interceptors.inc:375
    #1 0x51a8f4 in json_include C:\Pandu\GitHub\c-core\core\pubnub_coreapi.c:274
    #2 0x519b4d in pubnub_set_state C:\Pandu\GitHub\c-core\core\pubnub_coreapi.c:349
    #3 0x4e162b in main C:\Pandu\GitHub\c-core\core\samples\pubnub_sync_sample.c:147
    #4 0x558583 in _scrt_common_main_seh d:\a01\_work\10\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
    #5 0x7673fa28  (C:\Windows\System32\KERNEL32.dll+0x6b81fa28)
    #6 0x77957a7d  (C:\Windows\SYSTEM32\ntdll.dll+0x4b2e7a7d)
    #7 0x77957a4d  (C:\Windows\SYSTEM32\ntdll.dll+0x4b2e7a4d)

0x0350308d is located 0 bytes to the right of 77-byte region [0x03503040,0x0350308d)
allocated by thread T0 here:
    #0 0x502f95 in malloc D:\a01\_work\10\s\src\vctools\crt\asan\llvm\compiler-rt\lib\asan\asan_malloc_win.cpp:118
    #1 0x51995e in pubnub_set_state C:\Pandu\GitHub\c-core\core\pubnub_coreapi.c:340
    #2 0x4e162b in main C:\Pandu\GitHub\c-core\core\samples\pubnub_sync_sample.c:147
    #3 0x558583 in _scrt_common_main_seh d:\a01\_work\10\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
    #4 0x7673fa28  (C:\Windows\System32\KERNEL32.dll+0x6b81fa28)
    #5 0x77957a7d  (C:\Windows\SYSTEM32\ntdll.dll+0x4b2e7a7d)
    #6 0x77957a4d  (C:\Windows\SYSTEM32\ntdll.dll+0x4b2e7a4d)

SUMMARY: AddressSanitizer: heap-buffer-overflow D:\a01\_work\10\s\src\vctools\crt\asan\llvm\compiler-rt\lib\sanitizer_common\sanitizer_common_interceptors.inc:375 in __asan_wrap_strlen
Shadow bytes around the buggy address:
  0x306a05c0: fa fa fd fd fd fd fd fd fd fd fd fa fa fa fa fa
  0x306a05d0: fd fd fd fd fd fd fd fd fd fd fa fa fa fa fd fd
  0x306a05e0: fd fd fd fd fd fd fd fa fa fa fa fa fd fd fd fd
  0x306a05f0: fd fd fd fd fd fa fa fa fa fa 00 00 00 00 00 00
  0x306a0600: 00 00 00 05 fa fa fa fa 00 00 00 00 00 00 00 00
=>0x306a0610: 00[05]fa fa fa fa fd fd fd fd fd fd fd fd fd fd
  0x306a0620: fa fa fa fa fd fd fd fd fd fd fd fd fd fd fa fa
  0x306a0630: fa fa fd fd fd fd fd fd fd fd fd fd fa fa fa fa
  0x306a0640: fd fd fd fd fd fd fd fd fd fd fa fa fa fa fd fd
  0x306a0650: fd fd fd fd fd fd fd fd fa fa fa fa fd fd fd fd
  0x306a0660: fd fd fd fd fd fa fa fa fa fa fd fd fd fd fd fd
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==8088==ABORTING

**Review by budgetpreneur — CHANGES_REQUESTED** [review]
Posted sanitizer reported issue.

**Comment by budgetpreneur** [comment]
Another thing noticed is state param data is incorrect.

**Pull request closed without merging**